taprpc/universerpc/marshal.go

```diff
@@ -0,0 +1,45 @@
+package universerpc
+
+import (
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/asset"
+)
+
+// MarshalOutpoint marshals a wire.OutPoint into an RPC ready Outpoint.
+//
+// TODO(ffranr): Move this package's Outpoint type and this marshal function to
+// somewhere more general.
+func MarshalOutpoint(outPoint wire.OutPoint) Outpoint {
+	return Outpoint{
+		HashStr: outPoint.Hash.String(),
+		Index:   int32(outPoint.Index),
+	}
+}
+
+// MarshalAssetKey returns an RPC ready AssetKey.
+func MarshalAssetKey(outPoint wire.OutPoint,
+	scriptKeyPubKey *btcec.PublicKey) AssetKey {
+
+	scriptKeyBytes := scriptKeyPubKey.SerializeCompressed()
+	rpcOutPoint := MarshalOutpoint(outPoint)
```

Comment:
If we return a pointer from these helper functions directly, we can use them inline and don't need to create another variable. Makes things more compact and IMO a bit easier to read. And in gRPC land, things are _always_ pointers of structs, so might as well return them as that in specific marshal functions.

Reply:
I'm happy to go down that route because it seems to be a common pattern in our code. But I want to say that there's something clearly wrong somewhere (with Go or otherwise) when we choose pointers for the sake of convenience and not for their function. Normally in most languages I would avoid pointers as much as possible.